### PR TITLE
MM-29488: Fix racy test TestScheduler

### PR DESCRIPTION
**Pull request by agnivade**
We iterate the nextRunTimes slice only after stopping the
scheduler.

And since we are checking the slice 3 times, we split
the test into 3 sub-tests, every time starting a new scheduler
and running the same steps again.

https://mattermost.atlassian.net/browse/MM-29488

```release-note
NONE
```

### CODE REVIEW

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

File: jobs/schedulers.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func (schedulers *Schedulers) scheduleJob(cfg *model.Config, scheduler model.Sch
 	return scheduler.ScheduleJob(cfg, pendingJobs, lastSuccessfulJob)
 }
 
-func (schedulers *Schedulers) handleConfigChange(oldConfig *model.Config, newConfig *model.Config) {
+func (schedulers *Schedulers) handleConfigChange(_ *model.Config, newConfig *model.Config) {
```
Comment (wiggin77) [inline jobs/schedulers.go:200]: Have we decided to start doing this?  I feel like it reduces the documentation of the method just to save a pointer copy.  In this case a reader might guess the first arg was a pointer to the old config, but in other methods it is not clear and someone may have to lookup the interface just to know what that arg was.  Also I think this is something to be left for the compiler to optimize.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: jobs/schedulers.go (hunk 1 above)
Comment (agnivade) [reply jobs/schedulers.go:200]: Oh it's not a performance optimization at all. This is just to clarify that the first argument is not used at all. So we don't pass what we don't use.

It's interesting that you see that it reduces the documentation. I can change it.

**Pull request merged**